go/engine/list_tracking.go

```diff
@@ -88,13 +94,24 @@ func (e *ListTrackingEngine) Run(m libkb.MetaContext) (err error) {
 		WithStubMode(libkb.StubModeUnstubbed).
 		WithCachedOnly(e.arg.CachedOnly).
 		WithSelf(uid.Exists() && uid.Equal(m.G().GetMyUID()))
+	if e.arg.CachedOnly && e.arg.CachedOnlyStalenessWindow != nil {
+		larg = larg.WithStaleOK(true)
+	}
 	upak, _, err := m.G().GetUPAKLoader().LoadV2(larg)
 	if err != nil {
 		return err
 	}
 	if upak == nil {
 		return libkb.UserNotFoundError{}
 	}
+
+	if e.arg.CachedOnly && e.arg.CachedOnlyStalenessWindow != nil {
+		if m.G().Clock().Since(keybase1.FromTime(upak.Uvv.CachedAt)) > *e.arg.CachedOnlyStalenessWindow {
+			msg := fmt.Sprintf("upak was cached but exceeded custom staleness window %v", *e.arg.CachedOnlyStalenessWindow)
+			return libkb.UserNotFoundError{UID: uid, Msg: msg}
```

Comment:
I feel like an OfflineError here would represent the failure better. Not sure if anything up the stack depends on this error type though.

Reply:
Yeah it's not perfect, but it's what upak_loader returns if it's too stale so I just kept it the same error type